lib/rubygems/package.rb

```diff
@@ -239,12 +239,14 @@ def add_metadata tar # :nodoc:
   ##
   # Builds this package based on the specification set by #spec=
 
-  def build skip_validation = false
+  def build skip_validation = false, allow_warnings = true
```

Comment:
You can use [this test](https://github.com/rubygems/rubygems/blob/master/test/rubygems/test_gem_package.rb#L282-L293) as a base for testing `package.build false, true`